circus/commands/kill.py

```diff
@@ -0,0 +1,109 @@
+from circus.commands.base import Command
+from circus.exc import ArgumentError, MessageError
+from circus.util import to_signum
+from tornado import gen
+
+
+class Kill(Command):
+    """\
+        Kill a specific process
+        =======================
+
+        This command allows you to terminate a process in a watcher. If a
+        process does not exit within graceful_timeout it will be terminated
+        with a SIGKILL.
+
+        ZMQ Message
+        -----------
+
+        To kill all the processes of a watcher::
+
+            {
+                "command": "kill",
+                "property": {
+                    "name": <name>,
+                    "signum": <signum>,
+                    "graceful_timeout": <graceful_timeout>
+                }
+            }
+
+        To send a signal to a specific process::
+
+            {
+                "command": "kill",
+                "property": {
+                    "name": <name>,
+                    "pid": <processid>,
+                    "signum": <signum>,
+                    "graceful_timeout": <graceful_timeout>
+                }
+            }
+
+
+        Command line
+        ------------
+
+        ::
+
+            $ circusctl kill <name> [<pid>] [<signum>] [<graceful_timeout>]
+
+        Options:
+        ++++++++
+
+        - <name>: the name of the watcher
+        - <pid>: integer, the process id
+        - <signum>: overrides the watcher's stop_signal (number or name)
+        - <graceful_timeout>: overrides the watcher's graceful_timeout
```

Comment:
I think `signum` and `graceful_timeout` should be options (`--signum` and `--graceful-timeout`). That way you could stop all the processes of a watcher with a specific signal for example.